src/sparsezoo/v2/utils/backwards_compatibility.py

```diff
@@ -0,0 +1,148 @@
+    request_json.append(onnx_model_file_dict)
+
+    # if NLP model, add `config.json` and `tokenizer.json` to `deployment`
+    training_file_names = [
+        x[1]["display_name"]
+        for x in fetch_from_request_json(request_json, "file_type", "training")
+    ]
+    nlp_folder = (
+        True
+        if (("config.json") in training_file_names)
+        and (("tokenizer.json") in training_file_names)
+        else False
+    )
+    if nlp_folder:
+        for file_name in ["config.json", "tokenizer.json"]:
+            file_dict_training = fetch_from_request_json(
+                request_json, "display_name", file_name
+            )
+            assert len(file_dict_training) == 1
+            file_dict_deployment = copy.copy(file_dict_training[0][1])
+            file_dict_deployment["file_type"] = "deployment"
+            request_json.append(file_dict_deployment)
+
+    # create recipes
+    recipe_dicts_list = fetch_from_request_json(request_json, "file_type", "recipe")
+    for (idx, file_dict) in recipe_dicts_list:
+        display_name = file_dict["display_name"]
+        # make sure that recipe name has a
+        # format `recipe_{...}`.
+        prefix = "recipe_"
+        if not display_name.startswith(prefix):
+            display_name = prefix + display_name
+            file_dict["display_name"] = display_name
+            request_json[idx] = file_dict
+
+    # restructure inputs/labels/originals/outputs directories
+    # use `sample-inputs.tar.gz` to simulate non-existent directories
+
+    files_to_create = [
+        "sample_inputs.tar.gz",
+        "sample_labels.tar.gz",
+        "sample_originals.tar.gz",
+        "sample_outputs.tar.gz",
+    ]
+    types = ["inputs", "labels", "originals", "outputs"]
+    for file_name, type in zip(files_to_create, types):
+        data = fetch_from_request_json(
+            request_json, "display_name", file_name.replace("_", "-")
+        )
+        if len(data) == 1:
+            # file present but needs
+            # restructuring
+            file_dict = data[0][1]
+            file_dict["display_name"] = file_name
+            file_dict["file_type"] = type
+            idx = data[0][0]
+            request_json[idx] = file_dict
+
+    # remove all undesired or duplicate files
+    request_json = [
+        file_dict
+        for file_dict in request_json
+        if file_dict["file_type"] in allowed_file_types
+    ]
+
+    return request_json
+
+
+def fetch_from_request_json(
+    request_json: List[Dict[str, Any]], key: str, value: str
+) -> List[Tuple[int, Dict[str, Any]]]:
+    # searches through the `request_json` list to find a
+    # dictionary, that contains the requested
+    # key-value pair.
+    # return a list of tuples
+    # (every tuple is a file_dict, together
+    # with the respective list index)
+    return [
+        (idx, copy.copy(file_dict))
+        for (idx, file_dict) in enumerate(request_json)
+        if file_dict[key] == value
```

Comment:
I have a suggestion, instead of using plain old tuples, maybe use `namedtuple` here, the idea being it works like a tuple, so your existing code will work + wherever we are using indexing we can replace that with named access

for example
```python
# using tuple
snorlax = ("Snorlax", 1_000_000)
weight = snorlax[1] # obscure index 1
```

with named tuple

```python
# with named tuple this becomes more readable

from collections import namedtuple
Something = namedtuple("Something", "name weight")

snorlax = Something(name='Snorlax', weight=1_000_000,)
weight = snorlax.weight # avoids obscure index
```